core/base/topoMap/TopoMap.h

```diff
@@ -0,0 +1,1045 @@
+/// \ingroup base
+/// \class ttk::TopoMap
+/// \author Alexandre Talon <alexandre.talon@lip6.fr>
+/// \date March 2023
+///
+///
+/// This module defines the %TopoMap class that embeds points into 2D. The input
+/// is a distance matrix representing the distances between any pair of points
+/// in high dimension. It computes the 2D coordinates of a projection of the
+/// points which preserves the distances of a minimum spanning tree computed by
+/// Kruskal's algorithm. It also preserves the components Kruskal's algorithm
+/// builds iteratively.
+///
+///
+/// \b Related \b publication: \n
+/// "Topomap: A 0-dimensional homology preserving projection of high-dimensional
+/// data"\n Harish Doraiswamy, Julien Tierny, Paulo J. S. Silva, Luis Gustavo
+/// Nonato, and Claudio Silva\n Proc. of IEEE VIS 2020.\n IEEE Transactions on
+/// Visualization and Computer Graphics 27(2): 561-571, 2020.
+
+/// \sa TopoMap
+
+#pragma once
+
+// ttk common includes
+#include <Debug.h>
+#include <Geometry.h>
+#include <UnionFind.h>
+
+// STL includes
+#include <map>
+#include <queue>
+#include <set>
+#include <sstream>
+#include <utility> // For std::pair
+#include <vector>
+
+// cos and sin computations, so we are cautious on testing equalities between
+// double.
+static double Epsilon{ttk::Geometry::pow(10.0, -FLT_DIG + 2)};
+static double EpsilonDBL{ttk::Geometry::pow(10.0, -DBL_DIG + 2)};
+
+template <typename T>
+static inline T compute_dist2(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return dx * dx + dy * dy;
+}
+
+template <typename T>
+static inline T compute_dist(const T *ptA, const T *ptB) {
+  T dx = ptB[0] - ptA[0], dy = ptB[1] - ptA[1];
+
+  return sqrt(dx * dx + dy * dy);
+}
+
+template <typename T>
+inline bool are_colinear(const T *pptA, const T *pptB, const T *pptC) {
+  double ptA[2] = {pptA[0], pptA[1]}, ptB[2] = {pptB[0], pptB[1]},
+         ptC[2] = {pptC[0], pptC[1]};
+  return fabs(ptA[0] * (ptB[1] - ptC[1]) + ptB[0] * (ptC[1] - ptA[1])
+              + ptC[0] * (ptA[1] - ptB[1]))
+         <= EpsilonDBL;
+}
+
+// Normalizes a given vector.
+template <typename T>
+static void
+  computeUnitVector(const T *coordOrig, const T *coordDest, T *coordVect);
+
+// Computes the oriented angle BAC
+template <typename T>
+static double computeAngle(const T *ptA, const T *ptB, const T *ptC);
```

Comment:
please use instead `ttk::Geometry::angle()`

Reply:
New function angle2D, and angle2DUndirected (to obtain angle between 0 and 2pi).